browser/ui/sidebar/sidebar_utils.cc

```diff
@@ -158,4 +164,46 @@ absl::optional<SidebarItem> AddItemForSidePanelIdIfNeeded(Browser* browser,
   return absl::nullopt;
 }
 
+void SetLastlyUsedSidePanel(PrefService* prefs,
+                            absl::optional<SidePanelEntryId> id) {
+  BuiltInItemType type = BuiltInItemType::kNone;
+  if (id) {
+    switch (*id) {
+      case SidePanelEntryId::kReadingList:
+        type = BuiltInItemType::kReadingList;
+        break;
+      case SidePanelEntryId::kBookmarks:
+        type = BuiltInItemType::kBookmarks;
+        break;
+      case SidePanelEntryId::kPlaylist:
+        type = BuiltInItemType::kPlaylist;
+        break;
+      case SidePanelEntryId::kChatUI:
+        type = BuiltInItemType::kChatUI;
+        break;
+      default:
+        break;
+    }
+  }
+
+  prefs->SetInteger(kLastlyUsedBuiltInItemType, static_cast<int>(type));
+}
+
+absl::optional<SidePanelEntryId> GetLastlyUsedSidePanel(Browser* browser) {
+  PrefService* prefs = browser->profile()->GetPrefs();
+  BuiltInItemType type = static_cast<BuiltInItemType>(
+      prefs->GetInteger(kLastlyUsedBuiltInItemType));
+  if (type == BuiltInItemType::kNone) {
```

Comment:
Optional: Is this case needed?

Reply:
Yes, need this as `BuiltInItemType::kNone` is initial value for `kLastUsedBuiltInItemType` prefs.

Reply:
Won't it be handled by this check?

```cpp
if (!static_cast<BraveBrowser*>(browser)
           ->sidebar_controller()
           ->model()
           ->GetIndexOf(type)) {
    return absl::nullopt;
  }
```

Reply:
ahaaa you're right :) Done!

Reply:
Oops, I found a existing bug and added fix(https://github.com/brave/brave-core/pull/21022/commits/4cb5007e8a9ad22f522ba0034540b8aa94bcf18b)